ale/drivers/isis_spice_driver.py

```diff
@@ -0,0 +1,588 @@
+    field names and the values are lists of entries.
+
+    Parameters
+    ----------
+    table_label : PVLModule
+        The table label
+
+    data : bytes
+        The binary data for the entire table
+
+    Returns
+    -------
+    dict
+      The table as a dict
+    """
+    fields = table_label.getlist('Field')
+    results = {field['Name']:[] for field in fields}
+    offset = 0
+    for record in range(table_label['Records']):
+        for field in fields:
+            field_data = parse_field(field, data[offset:])
+            results[field['Name']].append(field_data)
+            offset += field_size(field)
+    return results
+
+def parse_rotation_table(label, field_data):
+    """
+    Parses ISIS rotation table data.
+
+    Parameters
+    ----------
+    table_label : PVLModule
+        The table label
+
+    field_data : dict
+        The table data as a dict with field names as keys
+        and lists of entries as values
+
+    Returns
+    -------
+    dict
+      The rotation data
+    """
+    results = {}
+    if all (key in field_data for key in ['J2000Q0','J2000Q1','J2000Q2','J2000Q3']):
+        results['Rotations'] = quaternion.as_quat_array( [ [q0, q1, q2, q3] for q0, q1, q2, q3 in zip(field_data['J2000Q0'],field_data['J2000Q1'],field_data['J2000Q2'],field_data['J2000Q3']) ] )
+    if all (key in field_data for key in ('AV1','AV2','AV3')):
+        results['AngularVelocities'] = np.array( [ [av1, av2, av3] for av1, av2, av3 in zip(field_data['AV1'],field_data['AV2'],field_data['AV3']) ] )
+    if 'ET' in field_data:
+        results['Times'] = np.array(field_data['ET'])
+    if all (key in field_data for key in ('J2000Ang1','J2000Ang2','J2000Ang3')):
+        results['EulerCoefficients'] = np.array([field_data['J2000Ang1'],field_data['J2000Ang2'],field_data['J2000Ang3']])
+        results['BaseTime'] = field_data['J2000Ang1'][-1]
+        results['TimeScale'] = field_data['J2000Ang2'][-1]
+
+    if 'TimeDependentFrames' in label:
+        results['TimeDependentFrames'] = np.array(label['TimeDependentFrames'])
+    if all (key in label for key in ('ConstantRotation','ConstantFrames')):
+        const_rotation_mat = np.array(label['ConstantRotation'])
+        results['ConstantRotation'] = quaternion.from_rotation_matrix(np.reshape(const_rotation_mat, (3, 3)))
+        results['ConstantFrames'] = np.array(label['ConstantFrames'])
+    if all (key in label for key in ('PoleRa','PoleDec','PrimeMeridian')):
+        results['BodyRotationCoefficients'] = np.array( [label['PoleRa'],label['PoleDec'],label['PrimeMeridian']] )
+    if all (key in label for key in ('PoleRaNutPrec','PoleDecNutPrec','PmNutPrec','SysNutPrec0','SysNutPrec1')):
+        results['SatelliteNutationPrecessionCoefficients'] = np.array( [label['PoleRaNutPrec'],label['PoleDecNutPrec'],label['PmNutPrec']] )
+        results['PlanetNutationPrecessionAngleCoefficients'] = np.array( [label['SysNutPrec0'],label['SysNutPrec1']] )
+    return results
```

Comment:
@Kelvinrr Any thoughts on how this might be cleaned up some so it is not a block of if/if/if/if searching for keys. Feels like something cleaner exists, but it is escaping me.

Reply:
I can't imagine anything much better, there may exists a compelling argument for taking advantage of the existing driver structure to use the different table types as mixins, but that's really just gonna shuffle the logic around. I feel in this case if blocks are fine, maybe just make the clauses easier to read (e.g. `{"key1", "key2"} <= label.keys()` assuming whatever `label` is formatted like affords it, otherwise just your above suggestion would be fine)